packages/backend-core/src/objectStore/objectStore.ts

```diff
@@ -26,12 +28,29 @@ type ListParams = {
 type UploadParams = {
   bucket: string
   filename: string
-  path: string
+  path?: string | PathLike
+  type?: string | null
+  // can be undefined, we will remove it
+  metadata?: {
+    [key: string]: string | undefined
+  }
+  body?: ReadableStream | Buffer
+  addTTL?: boolean
```

Comment:
Is there any reason why this is a boolean instead of specifying the TTL? Or is this because we cannot have different TTLs per object?